concordium-consensus/src/Concordium/GlobalState/Persistent/Bakers.hs

```diff
@@ -72,10 +74,14 @@ instance (IsAccountVersion av, MonadBlobStore m) => BlobStorable m (BakerInfos a
         return $ BakerInfos <$> sequence v
 
 -- | This hashing should match (part of) the hashing for 'Basic.EpochBakers'.
-instance (IsAccountVersion av, MonadBlobStore m) => MHashableTo m H.Hash (BakerInfos av) where
-    getHashM (BakerInfos v) = do
-        v' <- mapM loadPersistentBakerInfoRef v
-        return $ H.hashLazy $ runPutLazy $ mapM_ put v'
+instance forall pv m. (IsProtocolVersion pv, MonadBlobStore m) => MHashableTo m H.Hash (BakerInfos pv) where
+    getHashM (BakerInfos infos) = do
+        loadedInfos <- mapM loadPersistentBakerInfoRef infos
+        case sBlockHashVersionFor (protocolVersion @pv) of
+            SBlockHashVersion0 -> do
+                return $ H.hashLazy $ runPutLazy $ mapM_ put loadedInfos
+            SBlockHashVersion1 -> do
+                return $ hashAsLFMBTV1 (H.hash "NoBakerInfos") $ H.hashLazy . runPutLazy . put <$> Vec.toList loadedInfos
```

Comment:
I'm not sure if this hashLazy here is beneficial. Isn't a bakerinfo quite a small structure? So it's probably better to just serialize and then do one hash FFI call.

Reply:
Not sure when to consider it a small structure.
It includes the baker ID, all the baker public keys and the pool info which is open status, metadata url and commission rates.
I can change it if you want

Reply:
In order to know what is actually best, we need benchmarks, for now this will most likely perform similar to before, so I won't do anything.